File: 1854-maximum-population-year/1854-maximum-population-year.py
class Solution:
    def maximumPopulation(self, logs: List[List[int]]) -> int:
        years = [0] * 101 
        for birth, death in logs:
            years[birth - 1950] += 1
            years[death - 1950] -= 1

        max_pop = 0
        max_year = 1950
        current_pop = 0

        for i in range(101):
            current_pop += years[i]
            if current_pop > max_pop:
                max_pop = current_pop
                max_year = 1950 + i

        return max_year
            

File: 1854-maximum-population-year/1854-maximum-population-year.cpp
class Solution {
public:
    int maximumPopulation(vector<vector<int>>& logs) {
        vector<int>populationByYear(101,0);
        for(int i=0;i<logs.size();i++)
        {
            populationByYear[logs[i][0]-1950]++;
            populationByYear[logs[i][1]-1950]--;
        }

        int maximumPopulation=0;
        int maximumPopulationyear=1950;
        int currentPopulation=0;
        for(int i=0;i<101;i++)
        {
            currentPopulation+=populationByYear[i];
            if(currentPopulation>maximumPopulation)
            {
                maximumPopulation=currentPopulation;
                maximumPopulationyear=i+1950;
            }
        }
        return maximumPopulationyear;
    }
};